lib/Workflow/Factory.pm

```diff
@@ -304,11 +304,18 @@ sub _load_observers {
                 "Workflow::Factory docs for details.)";
         }
     }
-    $log->is_info
-        && $log->info( "Added observers to '$wf_type': ", join ', ',
-        @observers );
-    $self->{_workflow_observers}{$wf_type}
-        = scalar @observers ? \@observers : undef;
+    # FIXME: is logged even though no observers are added
```

Comment:
Didn't you fix this below? I mean `if (scalar @observers) {}` executes its conditional block only if `scalar @observers` is non-zero.

BTW, did you know that `if (scalar @observers)` is the same as `if (@observers)` since empty arrays are "false" too?